app/src/main/java/com/example/atheneum/activities/AddBookActivity.java

```diff
@@ -224,7 +168,35 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
             for (EditTextWithValidator editTextVal : editTextWithValidatorArray) {
                 editTextVal.validator.validate(editTextVal.editText);
             }
+            return;
+        }
 
+        Log.i(TAG, "AddBook*** Fields checked, all filled");
+
+        title = titleEditText.getText().toString();
+        author = authorEditText.getText().toString();
+        isbn = Long.parseLong(isbnEditText.getText().toString());
+        desc = descEditText.getText().toString();
+
+        if (FirebaseAuthUtils.isCurrentUserAuthenticated()) {
+            FirebaseUser firebaseUser = FirebaseAuthUtils.getCurrentUser();
+            AddBookViewModelFactory factory = new AddBookViewModelFactory(firebaseUser.getUid());
+            final AddBookViewModel addBookViewModel = ViewModelProviders.of(AddBookActivity.this, factory).get(AddBookViewModel.class);
```

Comment:
I get a crash on this line when trying to add a book. 
```
2019-03-05 07:57:28.315 3882-3882/com.example.atheneum E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.atheneum, PID: 3882
    java.lang.ClassCastException: com.example.atheneum.viewmodels.UserViewModel cannot be cast to com.example.atheneum.viewmodels.AddBookViewModel
        at com.example.atheneum.activities.AddBookActivity.saveNewBook(AddBookActivity.java:184)
        at com.example.atheneum.activities.AddBookActivity$1.onClick(AddBookActivity.java:110)
        at android.view.View.performClick(View.java:6256)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

Reply:
Looks like AddBookViewModelFactory.java has it returning a UserViewModel rather than the AddBookViewModel

Reply:
Fixed, thanks!